Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Quality/QualityCVarGroup.h>
+#include <AzFramework/Quality/QualitySystemBus.h>
+
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AzFramework
+{
+    QualityCVarGroup::QualityCVarGroup(AZ::SettingsRegistryInterface* registry, AZ::IConsole* console, AZStd::string_view name, AZStd::string_view path)
+        : m_console(console)
+        , m_numQualityLevels(0)
+        , m_path(path)
+        , m_qualityLevel(0)
+        , m_settingsRegistry(registry)
+    {
+        AZ_Assert(registry, "QualityCVarGroup requireds a valid settings registry.");
+        AZ_Assert(console, "QualityCVarGroup requireds a valid settings registry.");
+
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+
+        // optional default quality level
+        registry->Get(m_qualityLevel, FixedValueString::format("%.*s/Default", AZ_STRING_ARG(path)));
+
+        // count the number of quality levels for error checking
+        auto callback = [&]([[maybe_unused]] const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+        {
+            m_numQualityLevels++;
+            return AZ::SettingsRegistryInterface::VisitResponse::Continue;
+        };
+        auto levelsKey = FixedValueString::format("%.*s/Levels", AZ_STRING_ARG(path));
+        AZ::SettingsRegistryVisitorUtils::VisitArray(*registry, callback, levelsKey);
+
+        // optional description
+        FixedValueString description;
+        registry->Get(description, FixedValueString::format("%.*s/Description", AZ_STRING_ARG(path)));
+
+        m_functor = AZStd::make_unique<QualityCVarGroupFunctor>(
+            name.data(),
+            description.data(),
+            AZ::ConsoleFunctorFlags::DontReplicate,
+            AZ::AzTypeInfo<ValueType>::Uuid(),
+            *this,
+            &QualityCVarGroup::CvarFunctor);
+    }
+
+    QualityCVarGroup::~QualityCVarGroup()
+    {
+    }

```

Comment:
This can be `= default;` here
```suggestion
    QualityCVarGroup::~QualityCVarGroup() = default;

```